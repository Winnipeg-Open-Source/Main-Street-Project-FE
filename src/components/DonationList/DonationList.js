import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Text } from 'pcln-design-system';
import List from 'components/List';
import DonationCard from 'components/DonationCard';

function DonationList ({ className, isLoading, donations }) {
    return (
        <List className={className}>
            {donations && donations.length > 0
                ? donations.map(donation => <DonationCard {...donation} />)
                : <Text textAlign='center'>No donations yet...</Text>
            }
        </List>
    );
}

DonationList.displayName = 'DonationList';

DonationList.propTypes = {
    className: PropTypes.string,
    donations: PropTypes.array,
};

DonationList.defaultProps = {
    className: '',
};

export default memo(DonationList);
