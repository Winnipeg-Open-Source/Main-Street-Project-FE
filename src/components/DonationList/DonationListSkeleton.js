import React, { memo } from 'react';
import PropTypes from 'prop-types';
import List from 'components/List';
import DonationCard from 'components/DonationCard';

function DonationListSkeleton ({ className }) {
    return (
        <List className={className}>
            <DonationCard isLoading />
            <DonationCard isLoading />
            <DonationCard isLoading />
        </List>
    );
}

DonationListSkeleton.displayName = 'DonationListSkeleton';

DonationListSkeleton.propTypes = {
    className: PropTypes.string,
};

DonationListSkeleton.defaultProps = {
    className: '',
};

export default memo(DonationListSkeleton);
