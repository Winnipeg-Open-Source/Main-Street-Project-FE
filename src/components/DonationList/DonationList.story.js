import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import DonationList from './DonationList';

const props = {
    donations: [
        {
            id: 1,
            donationDate: '2020 Feb 02',
        },
        {
            id: 2,
            donationDate: '2020 Jan 30',
        },
    ],
};

export const WithoutDonations = () => <DonationList />;

export const WithDonations = () => (
    <BrowserRouter>
        <DonationList {...props} />
    </BrowserRouter>
);

export default {
    title: 'DonationList',
    component: DonationList,
};
