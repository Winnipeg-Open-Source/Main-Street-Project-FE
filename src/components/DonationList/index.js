import WithSkeleton from 'hocs/WithSkeleton';
import DonationList from './DonationList';
import DonationListSkeleton from './DonationListSkeleton';

export default WithSkeleton(DonationListSkeleton, DonationList);
