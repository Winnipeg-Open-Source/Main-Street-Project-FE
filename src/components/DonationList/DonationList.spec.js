import React from 'react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import DonationList from './DonationList';

jest.mock('components/DonationCard', () => () => 'Donation Card');

describe ('DonationList', () => {
    it ('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<DonationList />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('renders with donation cards', () => {
        const { getByText } = renderWithTheme(<DonationList donations={[{}]} />);

        expect(getByText('Donation Card')).toBeInTheDocument();
    });

    it ('renders without donation cards', () => {
        const { getByText } = renderWithTheme(<DonationList />);

        expect(getByText('No donations yet...')).toBeInTheDocument();
    });
});
