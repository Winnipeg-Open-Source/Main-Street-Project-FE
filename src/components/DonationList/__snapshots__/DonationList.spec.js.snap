// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationList renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pkfPC haHNKR sc-psDhf iAENMo"
  >
    <div
      class="sc-AxirZ sc-AxhCb sc-qQxFs iHzsts"
      width="1"
    >
      <div
        class="sc-AxhUy ilqMHx"
      >
        No donations yet...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
